### ISSUE TITLE
 BUG: KFENCE: use-after-free read in amdgpu_bo_move+0x1ce/0x710 [amdgpu]

### ISSUE THREAD

**Issue by kreutzi** [NixOS/nixpkgs#304090]
### Describe the bug
Error screen after suspend
```console
Apr 14 16:44:01 nixos kernel: ==================================================================
Apr 14 16:44:01 nixos kernel: BUG: KFENCE: use-after-free read in amdgpu_bo_move+0x1ce/0x710 [amdgpu]
Apr 14 16:44:01 nixos kernel: Use-after-free read at 0x0000000040d712ce (in kfence-#252):
Apr 14 16:44:01 nixos kernel: 
Apr 14 16:44:01 nixos kernel: 
Apr 14 16:44:01 nixos kernel: CPU: 10 PID: 1633 Comm: kwin_wayla:cs0 Tainted: P    B      O       6.6.26 #1-NixOS
Apr 14 16:44:01 nixos kernel: Hardware name: LENOVO 82JU/LNVNB161216, BIOS GKCN60WW 03/07/2023
Apr 14 16:44:01 nixos kernel: ==================================================================
Apr 14 20:18:24 nixos kernel: ==================================================================
Apr 14 20:18:24 nixos kernel: BUG: KFENCE: use-after-free read in amdgpu_bo_move+0x1ce/0x710 [amdgpu]
Apr 14 20:18:24 nixos kernel: Use-after-free read at 0x00000000362c5f46 (in kfence-#56):
Apr 14 20:18:24 nixos kernel: 
Apr 14 20:18:24 nixos kernel: 
Apr 14 20:18:24 nixos kernel: CPU: 3 PID: 186975 Comm: systemd-sleep Tainted: P    B      O       6.6.26 #1-NixOS
Apr 14 20:18:24 nixos kernel: Hardware name: LENOVO 82JU/LNVNB161216, BIOS GKCN60WW 03/07/2023
Apr 14 20:18:24 nixos kernel: ==================================================================
Apr 14 20:18:37 nixos kernel: ==================================================================
Apr 14 20:18:37 nixos kernel: BUG: KFENCE: use-after-free read in amdgpu_bo_move+0x1ce/0x710 [amdgpu]
Apr 14 20:18:37 nixos kernel: Use-after-free read at 0x0000000034ce0747 (in kfence-#163):
Apr 14 20:18:37 nixos kernel: 
Apr 14 20:18:37 nixos kernel: 
Apr 14 20:18:37 nixos kernel: CPU: 10 PID: 1633 Comm: kwin_wayla:cs0 Tainted: P    B      O       6.6.26 #1-NixOS
Apr 14 20:18:37 nixos kernel: Hardware name: LENOVO 82JU/LNVNB161216, BIOS GKCN60WW 03/07/2023
Apr 14 20:18:37 nixos kernel: ==================================================================

```

### Steps To Reproduce
This happened to me the 2 times my laptop was on suspend 
1. open firefox
2. suspend 
3. reopen laptop

### Screenshots

![Screenshot_20240414_202638](https://github.com/NixOS/nixpkgs/assets/88722623/40be3a64-b03e-4603-b695-b6a3b4a229ce)

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.26, NixOS, 24.05 (Uakari), 24.05.20240410.1042fd8`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/r1g09wfd968168f6drdl7agfaf3msik4-source`


```

---

**Comment by AkechiShiro** [comment]
@kreutzi could you precise your exact hardware and AMD gpu ? Have you looked at any issues upstream in https://gitlab.freedesktop.org/drm/amd/-/issues/ ?

**Comment by kreutzi** [comment]
@AkechiShiro 
CPU: AMD Ryzen 7 5800H with Radeon Graphics (16) @ 4.463GHz 
GPU: NVIDIA GeForce RTX 3060 Mobile / Max-Q 
GPU: AMD ATI Radeon Vega Series / Radeon Vega Mobile Serie

**Comment by AkechiShiro** [comment]
You'll probably need to apply the patch on your current kernel to fix the issue or wait until the 6.9/6.10 kernel is released.

https://lists.freedesktop.org/archives/amd-gfx/2024-March/105680.html
